app/models/namespace.rb

```diff
@@ -111,6 +130,88 @@ def ransackable_attributes(_auth_object = nil)
     def ransackable_associations(_auth_object = nil)
       %w[]
     end
+
+    def subtract_from_metadata_summary_count(namespaces, metadata_summary, by_one)
+      return if metadata_summary.empty?
+
+      update_metadata_summary_counts(namespaces, metadata_summary, by_one: by_one, addition: false)
+    end
+
+    def add_to_metadata_summary_count(namespaces, metadata_summary, by_one)
+      return if metadata_summary.empty?
+
+      update_metadata_summary_counts(namespaces, metadata_summary, by_one: by_one, addition: true)
+    end
+
+    private
+
```

Comment:
In the following section, the code should be refactored into a class called `PrivilegedNamespaceMetadataQuery` which can contain the complex `Arel` commands and have functions that return tables/ partial queries that get used in various places.